.github/workflows/create-release-branch.yml

```diff
@@ -0,0 +1,146 @@
+    steps:
+      - id: main
+        run: |
+          # If the workflow was manually triggered, then it should always be allowed to run to completion.
+          [[ "${{ github.event_name }}" = "workflow_dispatch" ]] && echo "should_run=true" >> "$GITHUB_OUTPUT" && exit 0
+          # `date -u` returns UTC datetime, and `%u` formats the output to be the day of the week, with 1 being Monday,
+          # 2 being Tuesday, etc.
+          TODAY_DOW=$(date -u +%u)
+          # This `date` expression returns the last Tuesday of the month, which is our Release Day. %d formats the output
+          # as the day of the month (1-31).
+          NEXT_RELEASE_DATE=$(date -u -v1d -v+1m -v-1d -v-tue +%d)
+          # This `date` expression returns next Tuesday, and `%d` formats the output as the day of the month (1-31).
+          NEXT_TUESDAY_DATE=$(date -u -v+tue +%d)
+          # If the workflow wasn't manually triggered, then it should only be allowed to run to completion on the Friday
+          # before Release Day.
+          [[ $TODAY_DOW != 5 || $NEXT_RELEASE_DATE != $NEXT_TUESDAY_DATE ]] && echo "should_run=false" >> "$GITHUB_OUTPUT" || echo "should_run=true" >> "$GITHUB_OUTPUT"
+  create-release-branch:
+    runs-on: macos-latest
+    needs: verify-should-run
+    if: ${{ needs.verify-should-run.outputs.should-run == 'true' }}
+    env:
+      GH_TOKEN: ${{ github.token }}
+    permissions:
+      contents: write
+    outputs:
+      branch-name: ${{ steps.create-branch.branch_name }}
+    steps:
+      # Checkout `dev`
+      - uses: actions/checkout@v4
+        with:
+          ref: 'dev'
+      # We need to set up Node and install our Node dependencies.
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*' # Always use Node LTS for building dependencies.
+      - run: yarn
+      # Increment the version as desired.
+      - name: Increment version
+        run: |
+          # A workflow dispatch event lets the user specify what release type they want.
+          if [[ "${{ github.event_name }}" = "workflow_dispatch" ]]; then
+            RELEASE_TYPE=${{ github.event.inputs.release-type }}
+          # The regularly scheduled releases are always minor.
+          else
+            RELEASE_TYPE=minor
+          fi
+          # Increment the version as needed
+          npm --no-git-tag-version version $RELEASE_TYPE
+      # Create the new branch as a direct copy of `dev` and push it to GitHub so the API can
+      # interact with it later.
+      - id: create-branch
+        run: |
+          NEW_VERSION=$(jq -r ".version" package.json)
+          git checkout -b release-$NEW_VERSION
+          # Push the branch immediately without committing anything.
+          git push --set-upstream origin release-$NEW_VERSION
+          # Output the branch name so we can use it in later jobs.
+          echo "branch_name=release-$NEW_VERSION" >> "$GITHUB_OUTPUT"
+      # Update our dependencies
+      - run: yarn upgrade
```

Comment:
I had some concerns about doing this, because in this repo we typically call `yarn install --frozen-lockfile` instead of just `yarn install`. However, upon further inspection, what that does is build without updating the `yarn.lock` file. So it's actually extra important to call `yarn upgrade` here because otherwise we're never going to update the yarn.lock.